doc/examples/uuid.rst

```diff
@@ -164,11 +165,48 @@ In this example, round-tripping the document using the incorrect
 ``PYTHON_LEGACY``) changes the :class:`~bson.binary.Binary` subtype as a
 side-effect. **Note that this can also happen when the situation is reversed -
 i.e. when the original document is written using ``STANDARD`` representation
-and then round-tripped using the ``PYTHON_LEGACY`` representation. Note also
-that replacing ``PYTHON_LEGACY`` by ``JAVA_LEGACY`` or ``CSHARP_LEGACY`` in
-the above example produces the same behavior.**
+and then round-tripped using the ``PYTHON_LEGACY`` representation.**
 
-.. note:: Starting in PyMongo 4.0, this issue will be resolved as
+In the next example, we see the consequences of incorrectly using a
+representation that modifies byte-order (``CSHARP_LEGACY`` or ``JAVA_LEGACY``)
+when round-tripping documents::
+
+  from bson.codec_options import CodecOptions, DEFAULT_CODEC_OPTIONS
+  from bson.binary import Binary, UuidRepresentation
+  from uuid import uuid4
+
+  # Using UuidRepresentation.STANDARD stores a Binary subtype-4 UUID
+  std_opts = CodecOptions(uuid_representation=UuidRepresentation.STANDARD)
+  input_uuid = uuid4()
+  collection = client.testdb.get_collection('test', codec_options=std_opts)
+  collection.insert_one({'_id': 'baz', 'uuid': input_uuid})
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 4)})['_id'] == 'baz'
+
+  # Retrieving this document using UuidRepresentation.JAVA_LEGACY returns a native UUID
+  # without modifying the UUID byte-order
+  java_opts = CodecOptions(uuid_representation=UuidRepresentation.JAVA_LEGACY)
+  java_collection = client.testdb.get_collection('test', codec_options=java_opts)
+  doc = java_collection.find_one({'_id': 'baz'})
+  assert doc['uuid'] == input_uuid
+
+  # Round-tripping the retrieved document silently changes the Binary bytes and subtype
+  java_collection.replace_one({'_id': 'baz'}, doc)
+  assert collection.find_one({'uuid': Binary(input_uuid.bytes, 3)}) is None
```

Comment:
I think this should be `Binary(input_uuid.bytes, 4)`

Reply:
Done - added the additional assert.